pkg/apiserver/webhooks/datavolume-validate.go

```diff
@@ -71,10 +71,7 @@ func validateDataVolumeName(name string) []metav1.StatusCause {
 }
 
 func validateContentTypes(sourcePVC *v1.PersistentVolumeClaim, spec *cdiv1.DataVolumeSpec) (bool, cdiv1.DataVolumeContentType, cdiv1.DataVolumeContentType) {
-	sourceContentType := cdiv1.DataVolumeKubeVirt
-	if contentType, found := sourcePVC.Annotations[controller.AnnContentType]; found {
-		sourceContentType = cdiv1.DataVolumeContentType(contentType)
-	}
+	sourceContentType := cdiv1.DataVolumeContentType(controller.GetContentType(sourcePVC))
```

Comment:
Do I understand it correctly thaT here we get DataVolumeKubeVirt, DataVolumeArchive or *empty*, 
and in the next lines for the targetContentType we get DataVolumeKubeVirt, DataVolumeArchive, and empty is not possible?

Reply:
@brybacki yep, we make sure targetContentType is not empty:
```
	targetContentType := spec.ContentType
	if targetContentType == "" {
		targetContentType = cdiv1.DataVolumeKubeVirt
	}

```

Reply:
> #1656 affects one of the base functionalities, we should have a tests for it,
> 
> the function func validateContentTypes(sourcePVC *v1.PersistentVolumeClaim, spec *cdiv1.DataVolumeSpec) (bool, cdiv1.DataVolumeContentType, cdiv1.DataVolumeContentType) can be covered by a unit tests with some source/target spec combinations

@brybacki good point. I was thinking all clone combinations were covered and all of them use this function, but may add the empty annotation as well.